src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1862,8 +1865,11 @@ GetRestrictInfoListForRelation(RangeTblEntry *rangeTblEntry,
 	List *joinRestrictClauseList = get_all_actual_clauses(joinRestrictInfo);
 	if (ContainsFalseClause(joinRestrictClauseList))
 	{
-		/* found WHERE false, no need  to continue */
-		return copyObject((List *) joinRestrictClauseList);
+		/* found WHERE false, no need  to continue, we just return a false clause */
+		bool value = false;
+		bool isNull = false;
+		List *falseClause = list_make1(makeBoolConst(value, isNull));
+		return falseClause;
```

Comment:
very minor: I was a bit surprised that falseClause was a list, maybe this can be `return list_make1(falseClause)`  and `falseClause` a `Const`